test/meta_test.js

```diff
@@ -119,4 +119,49 @@ suite('Repo Meta Tests', function() {
       }
     }
   });
+  test('headings in docs files match expectations', async function () {
+    const markdowns = glob.sync(
+      'ui/docs/**/*.md',
+      { cwd: ROOT_DIR });
+
+    for (let filename of markdowns) {
+      var data = fs.readFileSync(filename, 'utf8');
+      var md = data.toString();
+      
+      //remove the markdown code blocks which may include python # comment which can be confused with # markdown heading, as in, ui/docs/manual/using/s3-uploads.md
+      md = md.replace(/```[a-z]*[\s\S]*?\```/g, "");
+      var hd = [];
+
+      //hd[i] stores the number of headings with level i
+      hd[1] = md.match(/^# /gm);
+      hd[2] = md.match(/^## /gm);
+      hd[3] = md.match(/^### /gm);
+      hd[4] = md.match(/^#### /gm);
+      hd[5] = md.match(/^##### /gm);
+      hd[6] = md.match(/^###### /gm);
+
+      //counting levels of headings present and marking the top level 
+      var countLevel = 0, topLevelHd = 7;
+      for (var i = 1; i <= 6; i++) {
+        if (hd[i] != null && hd[i].length > 0) {
+          countLevel++;
+          if (i < topLevelHd)
+            topLevelHd = i;
+        }
+      }
+
+      // check if there is a single top-level heading
+      if (topLevelHd < 7) {
+        if (hd[topLevelHd].length > 1) {
+          throw new Error(`${filename} does not match the expectations`);
+        }
+      }
+
+      //check if there are no headings besides a single top-level heading
+      if (countLevel == 1) {
+        if (hd[topLevelHd].length == 1)
+          throw new Error(`${filename} does not match the expectations`);
```

Comment:
Could you add some more explanation to these two errors, so the user knows what to fix?